### ISSUE TITLE
[Settings] Create a route.ts file to handle populating Settings

### ISSUE THREAD

**Issue by psealock** [woothemes/woocommerce#52276]
Inspiration:

* [Product Editor](https://github.com/woocommerce/woocommerce/blob/trunk/packages/js/product-editor/src/products-app/router.tsx)
* [Site Editor](https://github.com/WordPress/gutenberg/blob/trunk/packages/edit-site/src/components/layout/router.js)
* [Proof of concept](https://github.com/woocommerce/woocommerce/blob/09cdb35a1ebbc422f87c5dea1795b0b7824d254e/plugins/woocommerce-admin/client/settings/routes.js)

## Criteria

* Routes should be represented as a JS object with the following properties. Create a type for this
```js
{
	key: <id>
	areas: {
		content: <MyExample section={ section } />,
		edit: <MyExampleEdit />,
	},
	widths: {
		content: undefined,
		edit: 380,
	},
};
```

* Create a `useSettingsLocation` helper function if its useful
* Use Typescript
* if these are the first TS files, choose a location. Product Editor is a js package, does this make sense here?

**Comment by psealock** [comment]
@chihsuan it might be helpful to see the branch in https://github.com/woocommerce/woocommerce/pull/52691 because I change the shape of `window.wcSettings?.admin?.settingsPages` into something more useful. I also add `RouterProvider` so `@wordpress/router` functions can be used.

![Image](https://github.com/user-attachments/assets/8e7c895a-40ae-4afe-a476-0a80e721eb9a)

**Comment by chihsuan** [comment]
@psealock Thanks Paul! Yeah, I noticed that you made some changes to the shape of ` window.wcSettings?.admin?.settingsPages` in POC so I was referring to the POC branch. 🙂 

> I also add RouterProvider so @wordpress/router functions can be used.

That's great. I added a TODO in the code [here](https://github.com/woocommerce/woocommerce/pull/52692/files#diff-06fb71282016ee9c28b39bc998eba8c9956455524c8e987dd237b7520b2d5aceR156) that we can replace it with `@wordpress/router` once it's available.
